scene/animate-images-with-image-overlay/src/main/java/com/esri/samples/animate_images_with_image_overlay/AnimateImagesWithImageOverlayController.java

```diff
@@ -0,0 +1,208 @@
+      Point pointForImageFrame = new Point(-120.0724273439448, 35.131016955536694, SpatialReferences.getWgs84());
+      Envelope imageFrameEnvelope = new Envelope(pointForImageFrame, 15.09589635986124, -14.3770441522488);
+      scene.setInitialViewpoint(new Viewpoint(imageFrameEnvelope, camera));
+
+      // create a new tiled layer from the World Dark Gray Base REST service and set it as the scene's basemap
+      Basemap basemap = new Basemap(new ArcGISTiledLayer("https://services.arcgisonline" +
+        ".com/arcgis/rest/services/Canvas/World_Dark_Gray_Base/MapServer"));
+      scene.setBasemap(basemap);
+
+      // create a new elevation source from the Terrain3D REST service and set it as the scene's base surface
+      Surface surface = new Surface();
+      surface.getElevationSources().add(new ArcGISTiledElevationSource("https://elevation3d.arcgis" +
+        ".com/arcgis/rest/services/WorldElevation3D/Terrain3D/ImageServer"));
+      scene.setBaseSurface(surface);
+
+      // create and append an image overlay to the scene view
+      sceneView.getImageOverlays().add(new ImageOverlay());
+
+      // instantiate a new empty list to hold image frames
+      imageFrames = new ArrayList<>();
+      // get the image files from local storage as an unordered list
+      File[] imageFiles = new File(System.getProperty("data.dir"), "./samples-data/PacificSouthWest").listFiles();
+      // sort the list of image files
+      if (imageFiles != null) {
+        Arrays.sort(imageFiles);
+        // create an image with the given path and use it to create an image frame
+        for (File file : imageFiles) {
+          ImageFrame imageFrame = new ImageFrame(file.getAbsolutePath(), imageFrameEnvelope);
+          imageFrames.add(imageFrame);
+        }
+      }
+
+      // populate the frames combo box with values
+      framesComboBox.getItems().addAll("60 frames per second", "30 frames per second", "15 frames per second");
+      // open the sample at 15fps
+      framesComboBox.getSelectionModel().select(2);
+
+      startNewAnimationTimer();
+
+    } catch (Exception e) {
+      // on any exception, print the stack trace
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Create a new image frame from the image at the current index and add it to the image overlay.
+   */
+  private void addNextImageFrameToImageOverlay() {
+
+    // set image frame to image overlay
+    sceneView.getImageOverlays().get(0).setImageFrame(imageFrames.get(imageIndex));
+    // increment the index to keep track of which image to load next
+    imageIndex++;
+    // reset index once all files have been loaded
+    if (imageIndex == imageFrames.size())
+      imageIndex = 0;
+  }
+
+  /**
+   * Controls the opacity of the image overlay using the slider.
+   */
+  @FXML
+  private void changeImageOverlayOpacity() {
+    sceneView.getImageOverlays().get(0).setOpacity((float) opacitySlider.getValue());
```

Comment:
Same here, if you make a class member `ImageOverlay imageOverlay`, you can avoid interrogating the sceneView:
```suggestion
    imageOverlay.setOpacity((float) opacitySlider.getValue());
```